cmd/agent/install_script.sh

```diff
@@ -313,40 +307,30 @@ if [ "$OS" = "RedHat" ]; then
     $sudo_cmd yum -y --disablerepo='*' --enablerepo='datadog' install $dnf_flag "$agent_flavor" || $sudo_cmd yum -y install $dnf_flag "$agent_flavor"
 
 elif [ "$OS" = "Debian" ]; then
+    apt_trusted_d_keyring="/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg"
+    apt_usr_share_keyring="/usr/share/keyrings/datadog-archive-keyring.gpg"
 
-    printf "\033[34m\n* Installing apt-transport-https\n\033[0m\n"
+    printf "\033[34m\n* Installing apt-transport-https, curl and gnupg\n\033[0m\n"
     $sudo_cmd apt-get update || printf "\033[31m'apt-get update' failed, the script will not install the latest version of apt-transport-https.\033[0m\n"
-    $sudo_cmd apt-get install -y apt-transport-https
-    # Only install dirmngr if it's available in the cache
-    # it may not be available on Ubuntu <= 14.04 but it's not required there
-    cache_output=`apt-cache search dirmngr`
-    if [ ! -z "$cache_output" ]; then
-      $sudo_cmd apt-get install -y dirmngr
-    fi
+    $sudo_cmd apt-get install -y apt-transport-https curl gnupg
     printf "\033[34m\n* Installing APT package sources for Datadog\n\033[0m\n"
-    $sudo_cmd sh -c "echo 'deb https://${apt_url}/ ${apt_repo_version}' > /etc/apt/sources.list.d/datadog.list"
+    $sudo_cmd sh -c "echo 'deb [signed-by=${apt_usr_share_keyring}] https://${apt_url}/ ${apt_repo_version}' > /etc/apt/sources.list.d/datadog.list"
+
+    if [ ! -f $apt_usr_share_keyring ]; then
+        $sudo_cmd touch $apt_usr_share_keyring
+    fi
 
     for key in "${APT_GPG_KEYS[@]}"; do
-      for retries in {0..4}; do
-        $sudo_cmd apt-key adv --recv-keys --keyserver "${keyserver}" "${key}" && break
-        if [ "$retries" -eq 4 ]; then
-          ERROR_MESSAGE="ERROR
-  Couldn't fetch Datadog public key ${key}.
-  This might be due to a connectivity error with the key server
-  or a temporary service interruption.
-  *****
-  "
-          false
-        fi
-        printf "\033[33m\napt-key failed to retrieve Datadog's public key ${key}, retrying in 5 seconds...\n\033[0m\n"
-        sleep 5
-        if [ "$retries" -eq 1 ]; then
-          printf "\033[34mSwitching to backup keyserver\n\033[0m\n"
-          keyserver="${backup_keyserver}"
-        fi
-      done
+        $sudo_cmd curl -o "/tmp/${key}" "https://${keys_url}/${key}"
```

Comment:
Should we add the retry flag to keep the retrying behavior?
```suggestion
        $sudo_cmd curl --retry 5 -o "/tmp/${key}" "https://${keys_url}/${key}"
```

Reply:
Yeah, let's do that!